spec/models/chunk_spec.rb

```diff
@@ -113,20 +113,74 @@
           #ignore
         end
       end
-      it {
-        resp = HTTParty.get(
-          "#{storage_provider.storage_url}/#{subject.storage_container}/#{subject.object_path}",
-          headers: storage_provider.auth_header
-        )
-        expect(resp.response.code.to_i).to eq(200)
-        expect(resp.body).to eq(chunk_data)
-        subject.purge_storage
-        resp = HTTParty.get(
-          "#{storage_provider.storage_url}/#{subject.storage_container}/#{subject.object_path}",
-          headers: storage_provider.auth_header
-        )
-        expect(resp.response.code.to_i).to eq(404)
-      }
+
+      context 'StorageProviderException' do
+        context 'Not Found' do
+          it {
+            expect {
+              subject.storage_provider.delete_object(subject.storage_container, subject.object_path)
+            }.not_to raise_error
+
+            resp = HTTParty.get(
+              "#{storage_provider.storage_url}/#{subject.storage_container}/#{subject.object_path}",
+              headers: storage_provider.auth_header
+            )
+            expect(resp.response.code.to_i).to eq(404)
+
+            expect {
+              subject.purge_storage
+            }.not_to raise_error
+          }
+        end
+
+        context 'Other Exception' do
+          let(:fake_storage_provider) { FactoryBot.create(:storage_provider) }
```

Comment:
I'm not sure how this works. The storage_provider factory is populated randomly by Faker, including the `url_root` and `auth_uri`. Unless you crafted the VCR cassette by hand, this is going to try to make calls to non-existent hosts. I'm not sure what HTTP errors are thrown due to this, but it's not the authentication failure you were looking for. Maybe try fiddling with the `service_user` or `service_pass` of the `subject.storage_provider`? You want to see an auth failure recorded in your VCR cassette.